### PR TITLE
[Monitor] support schedule query rule

### PR DESCRIPTION
**Pull request by MyronFanQiu**
---

Fix Azure/azure-cli#14691

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S175

File: src/scheduled-query/azext_scheduled_query/azext_metadata.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "azext.isPreview": true,
```
Comment (jsntcy) [inline src/scheduled-query/azext_scheduled_query/azext_metadata.json:2]: isExperimental?

File: src/scheduled-query/azext_scheduled_query/_actions.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        lexer = ScheduleQueryConditionLexer(antlr4.InputStream(string_val))
+        stream = antlr4.CommonTokenStream(lexer)
+        parser = ScheduleQueryConditionParser(stream)
+        tree = parser.expression()
+
+        try:
+            validator = ScheduleQueryConditionValidator()
+            walker = antlr4.ParseTreeWalker()
+            walker.walk(validator, tree)
+            metric_condition = validator.result()
```
Comment (jsntcy) [inline src/scheduled-query/azext_scheduled_query/_actions.py:38]: scheduled_query_condition

File: src/scheduled-query/azext_scheduled_query/_help.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    short-summary: The condition which triggers the rule.
+    long-summary: |
+        Usage:  --condition {avg,min,max,total,count} ["METRIC COLUMN" from]
+                            "QUERY" {=,!=,>,>=,<,<=} THRESHOLD
+                            [resource id RESOURCEID]
+                            [where DIMENSION {includes,excludes} VALUE [or VALUE ...]
+                            [and   DIMENSION {includes,excludes} VALUE [or VALUE ...] ...]]
+                            [at least MinTimeToFail out of EvaluationPeriod]'
+
+        Dimensions can be queried by adding the 'where' keyword and multiple dimensions can be queried by combining them with the 'and' keyword.
```
Comment (jsntcy) [inline src/scheduled-query/azext_scheduled_query/_help.py:35]: Add examples

File: src/scheduled-query/README.rst [hunk 4]
```diff
@@ -0,0 +1,5 @@
+Microsoft Azure CLI 'scheduled-query' Extension
+==========================================
+
+This package is for the 'scheduled-query' extension.
+i.e. 'az scheduled-query'
```
Comment (jsntcy) [inline src/scheduled-query/README.rst:5]: Add something in readme?

**Review by jsntcy — APPROVED** [review]
:shipit:

File: src/scheduled-query/azext_scheduled_query/azext_metadata.json (hunk 1 above)
Comment (MyronFanQiu) [reply src/scheduled-query/azext_scheduled_query/azext_metadata.json:2]: I think `isPreview` is enough.

**Pull request merged**